src/plugins/support.js

```diff
@@ -0,0 +1,60 @@
+const path = require('path');
+const semver = require('semver');
+const pkgSupport = require('@pkgjs/support');
+const nv = require('@pkgjs/nv');
+
+const { stringBuilder, warning, success } = require('../lib/format');
+const { passThroughError } = require('../lib/result');
+
+const supportPlugin = async (pkg) => {
+  // Support plugin output
+  const output = stringBuilder('\nChecking for LTS support').withPadding(66);
+
+  const pkgPath = path.resolve(
+    process.cwd(),
+    `npcheck-env/node_modules/${pkg.name}`
+  );
+
+  const support = await pkgSupport.getSupportData(pkg, pkgPath);
+
+  const supportData = Buffer.isBuffer(support.contents)
+    ? JSON.parse(support.contents.toString())
+    : support.contents;
+
+  if (supportData !== 'unknown') {
+    // Check for LTS node target
+    const targetsLTS = supportData.versions?.find(
+      (item) => item?.target?.node === 'lts'
+    );
+    if (targetsLTS) {
+      success(output.get());
+      return null;
+    }
+  }
+
+  /* Couldn't detect LTS support from @pkg/support so let's try the engines field. */
+
+  const versions = await nv('lts');
+  const engines = pkg?.engines?.node || '0.0.0';
+
+  const unsupported = versions
+    .map((v) => v.version)
+    .map((v) => ({
+      version: v,
+      support: semver.satisfies(v, engines)
+    }))
+    .filter((v) => !v.support);
```

Comment:
The code only looks at `.version` and the length of the resulting `unsupported`, so we don't need to include `.support` in the `unsupported` result.

Replace with something like (untested):
```suggestion
  const unsupported = versions
    .map((v) => v.version)
    .filter((v) => !semver.satisifies(v, engines));
```
? Then the message passed to `passThroughError` below could be `The module "${pkg.name}" appears to have no support for the LTS version (v${unsupported[0]}) of Node.js.` or even something like `${unsupported.join(',')}` to include all unsupported versions in the message.

Reply:
Nice catch @richardlau, Thanks! I was also thinking about how I can print all unsupported LTS versions but I never thought of the `join` method. 🤦‍♂️